### PR TITLE
sphinx book theme

### LINKED ISSUE #49: Any interest in the sphinx-book-theme for docs?
We've been working on a simple sphinx theme for the jupyter-book rewrite that I've also found useful for other documentation sites as well (sphinx-book-theme.readthedocs.io)

I find the Alabaster theme to be a little bit cluttered and hard to parse sometimes...would folks be interested in a PR that updates the docs theme for nbclient to the sphinx-book-theme? Here's an example of what it looks like by just swapping the theme name in `conf.py`:

image

### PR DESCRIPTION
**Pull request by choldgraf**
This adds the sphinx-book-theme for our documentation. It also adds a CircleCI job to let us preview the built documentation in a PR (and removes the docs job in `tox` since it's redundant now)

closes #49 

here are the docs: https://4-233653381-gh.circle-artifacts.com/0/html/index.html

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -14,17 +14,6 @@ skip_install = true
 deps = check-manifest
 commands = check-manifest
 
-# Docs
```
Comment (MSeal) [inline tox.ini:17]: I would still like to keep this, as it let's me test that the docs are alright before making PRs

**Comment by choldgraf** [comment]
Just in the ini file or in Travis as well?

**Comment by choldgraf** [comment]
(I've just pushed a commit assuming you're OK with keeping it out of travis, but leaving it in the tox file, but lemme know if you'd like it in travis as well)

**Comment by MSeal** [comment]
That works for me, just having the local build helps when doing doc additions. I don't mind as much which tool automates the PR build aspect so long as some tool does.

**Comment by choldgraf** [comment]
sounds good - then lemme know if you'd like any extra changes 👍

**Comment by MSeal** [comment]
LGTM, merging

**Pull request merged**